src/net.cpp

```diff
@@ -2322,8 +2375,13 @@ bool CConnman::Start(CScheduler& scheduler, const Options& connOptions)
 
     if (!gArgs.GetBoolArg("-dnsseed", true))
         LogPrintf("DNS seeding disabled\n");
-    else
+    else {
         threadDNSAddressSeed = std::thread(&TraceThread<std::function<void()> >, "dnsseed", std::function<void()>(std::bind(&CConnman::ThreadDNSAddressSeed, this)));
+        if (fListen && fNameLookup && !HaveNameProxy()) {
+            LogPrintf("Proactive querying DNS servers to update caches is enabled\n");
+            scheduler.scheduleEvery([this] { DNSCachesUpdate(); }, DNS_CACHES_UPDATE_INTERVAL);
```

Comment:
Probably there should be some randomness added to the interval for each iteration?